src/Repository/AddressRepository.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Repository;
+
+use App\Dto\AddressRequest;
+use App\Entity\Address;
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
+
+class AddressRepository implements AddressRepositoryInterface
+{
+    private $entityManager;
+    private $repository;
+
+    public function __construct(EntityManagerInterface $entityManager)
+    {
+        $this->entityManager = $entityManager;
+        $this->repository = $entityManager->getRepository(Address::class);
+    }
+
+    public function save(Address $address): void
+    {
+        $this->entityManager->persist($address);
+        $this->entityManager->flush();
+    }
+
+    public function remove(Address $address): void
+    {
+        $this->entityManager->remove($address);
+        $this->entityManager->flush();
+    }
+
+    public function find(string $id): ?Address
+    {
+        return $this->repository->find($id);
+    }
+
+    public function findAll(): array
+    {
+        return $this->repository->findAll();
+    }
+
+    public function createFromRequest(AddressRequest $addressRequest): Address
+    {
+        $address = new Address();
+
+        return $address
+            ->setId($this->nextIdentity()->toString())
```

Comment:
if you remove the return $this from the entity you will need to change this part, IMO I would set everything into the constructor and remove the set methods that are not used other than creating the Entity, that way you are certain of the dependencies the Address Entity need, and it reduces possibilities of an incomplete Object.
Suppose another dev want to create a new Address but forgot to set the Id, the object will be created in the program and not being "valid", whereas if you put it into the constructor, the dev will know the dependencies needed to have a "valid" address and will b able to miss needed information.
To illustrate: https://matthiasnoback.nl/2018/07/objects-should-be-constructed-in-one-go/